src/main/scala/scala/tools/refactoring/sourcegen/TreeChangesDiscoverer.scala

```diff
@@ -156,9 +156,16 @@ trait TreeChangesDiscoverer {
           List((changed, orig.pos, Set(changed) ++ searchChildrenForChanges(changed)))
         }
 
+        def drillDownChildrenAndCollectChangesWithPosition(parent: Tree): List[(Tree, Position, Set[Tree])] =
```

Comment:
Hmm, I have trouble understanding this: Can you explain why the `TreeChangesDiscoverer` did not work correctly without this patch?